src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,94 @@
-export const Pagination = () => {};
+import { FC, useEffect, useState } from 'react';
+import cn from 'classnames';
+import { getNumbers } from '../../utils';
+
+type Props = {
+  total: number
+  perPage: number
+  currentPage: number
+  onPageChange: (page: number) => void
+};
+
+export const Pagination: FC<Props> = ({
+  total,
+  perPage,
+  currentPage = 1,
+  onPageChange,
+}) => {
+  const pages = getNumbers(1, Math.ceil(total / perPage));
+
+  const [prev, setPrev] = useState(currentPage - 1);
+  const [next, setNext] = useState(currentPage + 1);
+
+  useEffect(() => {
+    setNext(currentPage + 1);
+    setPrev(currentPage - 1);
+  }, [currentPage]);
+
+  const renderStepButton = (direction: 'prev' | 'next') => {
```

Comment:
- better to separate the type. If you'll need to change it - you will need to look for exactly this piece of code
- if your values type can only be strings - it is much better to use Enum for that. If it is used the way you have it now - the possibility of making a mistake is much higher